src/ophyd_async/epics/areadetector/writers/_hdffile.py

```diff
@@ -2,18 +2,25 @@
 
 from event_model import StreamDatum, StreamResource, compose_stream_resource
 
+from ophyd_async.core import DirectoryInfo
+
 from ._hdfdataset import _HDFDataset
 
 
 class _HDFFile:
-    def __init__(self, full_file_name: str, datasets: List[_HDFDataset]) -> None:
+    def __init__(
+        self,
+        directory_info: DirectoryInfo,
+        full_file_name: str,
+        datasets: List[_HDFDataset],
+    ) -> None:
         self._last_emitted = 0
         self._bundles = [
             compose_stream_resource(
                 spec="AD_HDF5_SWMR_SLICE",
-                root="/",
+                root=str(directory_info.root),
                 data_key=ds.name,
-                resource_path=full_file_name,
+                resource_path=str(directory_info.resource_dir / full_file_name),
```

Comment:
`full_file_name` is the complete path, if you want to separate out the root then you need `full_file_name` relative to `directory_info.root / directory_info.resource_dir`...

Reply:
Are we missing a test that would catch this too?

Reply:
Added a test as part of the change that should catch if the behaviour changes